src/Core/StackExpression.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Rokka\Client\Core;
+
+/**
+ * @since 1.1.0
+ */
+class StackExpression
+{
+    /**
+     * @var string
+     */
+    public $expression;
+
+    /**
+     * @var array
+     */
+    public $overrides;
+
+    public function __construct($expression, $optionOverrides = [])
+    {
+        $this->expression = $expression;
+        $this->overrides = ['options' => $optionOverrides];
+    }
+
+    /**
+     * Create a stack from the JSON data returned by the rokka.io API.
+     *
+     * @param string|array $data    JSON data
+     * @param bool         $isArray If the data provided is already an array
+     *
+     * @return StackExpression
+     */
+    public static function createFromJsonResponse($data, $isArray = false)
+    {
+        if (!$isArray) {
+            $data = json_decode($data, true);
+        }
+
+        $expression = new self($data['expression']);
+        $expression->overrides = $data['overrides'];
+
+        return $expression;
+    }
+
+    /**
+     * @return string
+     */
+    public function getExpression()
+    {
+        return $this->expression;
+    }
+
+    /**
+     * @param string $expression
+     *
+     * @return StackExpression
```

Comment:
Sometimes we have `return $this`, I am still unsure if the correct one is `@return self` as mentioned here: https://docs.phpdoc.org/references/phpdoc/types.html

Reply:
right, changed it, seems phpstorm does the $this thing but understands also the self thing.